polyA/calculate_score.py

```diff
@@ -70,5 +74,81 @@ def calculate_score(
                 chunk_score += gap_init
         else:
             chunk_score += sub_matrix[seq1_char + seq2_char]
-
+            if char_complexity_adjustment:
+                chunk_score += char_complexity_adjustment[seq2_char]
     return chunk_score
+
+
+def calc_target_char_counts(
+    query_seq: str,
+    target_seq: str,
+):
+    query_char_counts = {"A": 0, "C": 0, "G": 0, "T": 0}
+    other_chars = set()
+    total_chars = 0
+    for q_base, t_base in zip(query_seq, target_seq):
+        if t_base not in ["A", "G", "T", "C"]:
+            other_chars.add(t_base)
+            continue
+        if q_base == "-":
+            continue
+        query_char_counts[t_base] += 1
+        total_chars += 1
+    return query_char_counts, total_chars, other_chars
```

Comment:
`total_chars` is easy to compute from `query_char_counts` so we probably don't need to return it. You can get it with `sum(query_char_counts.values())`.